experimental/traffic-portal/src/app/core/statuses/status-details/status-details.component.ts

```diff
@@ -0,0 +1,132 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, OnInit } from "@angular/core";
+import { FormBuilder, FormControl, FormGroup, Validators } from "@angular/forms";
+import { MatDialog } from "@angular/material/dialog";
+import { ActivatedRoute, Router } from "@angular/router";
+import { ResponseStatus } from "trafficops-types";
+
+import { StatusesService } from "src/app/api/statuses.service";
+import { DecisionDialogComponent, DecisionDialogData } from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+
+@Component({
+	selector: "tp-status-details",
+	templateUrl: "./status-details.component.html",
+	styleUrls: ["./status-details.component.scss"]
+})
+export class StatusDetailsComponent implements OnInit {
+
+	id: string | null = null;
+	statusDetails: ResponseStatus | null = null;
+	statusDetailsForm!: FormGroup;
+	loading = false;
+	submitting = false;
+	submitted = false;
+
+	constructor(
+		private readonly route: ActivatedRoute,
+		private readonly router: Router,
+		private readonly fb: FormBuilder,
+		private readonly dialog: MatDialog,
+		private readonly statusesService: StatusesService) { }
+
+	ngOnInit(): void {
+		// Form is built here
+		this.statusDetailsForm = this.fb.group({
+			name: ["", Validators.required],
+			description: ["", Validators.required],
+		});
+
+		// Getting id from the route
+		this.id = this.route.snapshot.paramMap.get("id");
+
+		// we check whether params is a number if not we shall assume user wants to add a new status.
+		if (!this.isNew) {
+			this.loading = true;
+			this.statusDetailsForm.addControl("id", new FormControl(""));
+			this.statusDetailsForm.addControl("lastUpdated", new FormControl(""));
+			this.getStatusDetails();
+		}
+	}
+
+	/*
+   * Reloads the servers table data.
+   * @param id is the id passed in route for this page if this is a edit view.
+  */
+	async getStatusDetails(): Promise<void> {
+		const id = this.id as string; // id Type 'null' is not assignable to type 'string'
+		this.statusDetails = await this.statusesService.getStatuses(id);
+		const data: ResponseStatus = {
+			name: this.statusDetails.name,
+			description: this.statusDetails.description,
+			lastUpdated: new Date(),
+			id: this.statusDetails.id
+		};
+		this.statusDetailsForm.patchValue(data);
+		this.loading = false;
+	}
+
+	// On submitting the form we check for whether we are performing Create or Edit
+	onSubmit() {
+		if (this.isNew) {
+			this.createStatus();
+
+		} else {
+			this.updateStatus();
+		}
+	}
+
+	// For Creating a new status
+	createStatus() {
+		this.statusesService.createStatus(this.statusDetailsForm.value).then((res: any) => {
+			if (res) {
+				this.id = res?.id;
+				this.router.navigate([`/core/statuses/${this.id}`]);
```

Comment:
This should probably replace the current URL instead of adding to the history stack, since it's a view on the same object just in a different state

Reply:
yes, the view changes while the state here changes. We will be showing a delete button after creation and changing submit button lable from "create" to "update"

Reply:
i have changed this, redirected to statuses page after creation